### PR TITLE
consensus, eth: only start core engine if I'm a miner

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -106,15 +106,18 @@ type PoW interface {
...
-	Start(chain ChainReader, inserter func(block *types.Block) error) error
+	// Start starts the engine
+	Start(chain ChainReader, inserter func(types.Blocks) (int, error)) error
 
-	// Stop the engine
+	// Stop stops the engine
 	Stop() error
+
+	// IsStarted returns whether the engine is started
+	IsStarted() bool
```
Comment (markya0616) [inline consensus/consensus.go]: IsRunning()

Comment (markya0616) [inline consensus/consensus.go]: OK

File: consensus/istanbul/backend/backend.go [hunk 2]
```diff
@@ -154,25 +153,32 @@ func (sb *simpleBackend) Commit(proposal istanbul.Proposal, seals []byte) error
 		// TODO: how do we check the block is inserted correctly?
 		return nil
 	}
-
-	return sb.inserter(block)
+	// if I'm not a proposer, insert the block directly and broadcast NewCommittedEvent
```
Comment (tailingchen) [inline consensus/istanbul/backend/backend.go]:
```
	if _, err := sb.inserter(types.Blocks{block}); err!=nil{
            return err
        }
	msg := istanbul.NewCommittedEvent{
		Block: block,
 	}
 	go sb.eventMux.Post(msg)
 	return nil
```

Comment (markya0616) [inline consensus/istanbul/backend/backend.go]: OK

File: consensus/istanbul/events.go [hunk 3]
```diff
@@ -18,23 +18,32 @@ package istanbul
 
 import (
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 )
 
+// ConsensusDataEvent is posted  Istanbul consensus data
```
Comment (bailantaotao) [inline consensus/istanbul/events.go]: remove redundant blank
`ConsensusDataEvent is posted Istanbul consensus data`

Comment (yutelin) [inline consensus/istanbul/events.go]: ConsensusDataEvent is posted when sending Istanbul consensus data?

File: consensus/istanbul/events.go [hunk 4]
```diff
@@ -18,23 +18,32 @@ package istanbul
...
 	Target common.Address
 	// consensus message data
 	Data []byte
 }
 
+// NewCommittedEvent is posted when I'm not a proposer but
+// a block has been committed from Istanbul consensus.
+type NewCommittedEvent struct{ Block *types.Block }
+
+// RequestEvent is posted a proposal to start the consensus
```
Comment (yutelin) [inline consensus/istanbul/events.go]: not sure what it means.

File: consensus/istanbul/events.go [hunk 5]
```diff
@@ -18,23 +18,32 @@ package istanbul
...
 type RequestEvent struct {
 	Proposal Proposal
 }
 
+// MessageEvent is posted an Istanbul message
```
Comment (yutelin) [inline consensus/istanbul/events.go]: not sure what it means.

File: consensus/istanbul/events.go (hunk 3 above)
Comment (markya0616) [inline consensus/istanbul/events.go]: OK

File: consensus/istanbul/events.go (hunk 4 above)
Comment (markya0616) [inline consensus/istanbul/events.go]: RequestEvent is posted to start the consensus?

File: consensus/istanbul/events.go (hunk 5 above)
Comment (markya0616) [inline consensus/istanbul/events.go]: MessageEvent is posted for Istanbul engine communication?

**Pull request merged**